pallets/connectors/src/lib.rs

```diff
@@ -612,11 +865,108 @@ pub mod pallet {
 				Error::<T>::AssetNotFound
 			);
 
-			let general_index: GeneralCurrencyIndex<u128, T::GeneralCurrencyPrefix> =
-				CurrencyIdOf::<T>::try_into(currency)?;
+			let general_index: GeneralCurrencyIndexOf<T> = CurrencyIdOf::<T>::try_into(currency)?;
 
 			Ok(general_index.index)
 		}
+
+		/// Returns the local currency identifier from from its general index.
+		///
+		/// Assumes the currency to be registered in the `AssetRegistry`.
```

Comment:
How do you mean, "assumes"? Because it handles the error case when it is not in the registry so I wouldn't say it assumes anything

Reply:
Good catch. We can either entirely remove this comment or rephrase it to `requires`.